include/rocksdb/options.h

```diff
@@ -1369,6 +1378,7 @@ struct TraceOptions {
   // Specify trace sampling option, i.e. capture one per how many requests.
   // Default to 1 (capture every request).
   uint64_t sampling_frequency = 1;
+  uint64_t trace_filter_option = kTraceFilterNone;
```

Comment:
Lets change the name of `trace_filter_option` to just `filter`. 
And add a comment above it saying that filtering happens before sampling so that the uses are aware of the order.